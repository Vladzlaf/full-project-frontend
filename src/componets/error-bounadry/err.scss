// .wrap__btn--err {
//     display: flex;
//     align-items: center;
//     flex-direction: column;
//     justify-content: center;
//     cursor: pointer;
//     color: #fff;
//     background: #4c8aee;
//     position: absolute;
//     border: none;
//     font-size: 16px;
//     padding: 10px 15px;
// }

// .wrap__btn--err:hover {
//     color: #4c8aee;
//     background: #fff;
//     transition: 0.3s;
// }