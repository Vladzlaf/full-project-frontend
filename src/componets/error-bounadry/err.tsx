import ButtonComp from '../../componets/buttonCopm';
import {useEffect, useState} from 'react';

export const CreateError = () => {
  const [error, setError] = useState(false);

  const catchErr = () => {
    setError(true);
  }

  useEffect(() => {
    if (error) {
      throw new Error('shiiiiiiiiiiiit here we go again');
    }
  })
  return (
  <div className="wrap">
    <ButtonComp label='Click and you will see' onClick={catchErr} />
  </div>
  );
};
