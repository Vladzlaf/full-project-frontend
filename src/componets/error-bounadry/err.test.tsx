import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { CreateError } from './err';

describe('CreateError', () => {
  it('should throw an error when the button is clicked', () => {
    // Arrange
    const { getByText } = render(<CreateError />);
    
    // Act & Assert
    expect(() => fireEvent.click(getByText('Click and you will see'))).toThrowError(
      'shiiiiiiiiiiiit here we go again'
    );
  });

  it('should not throw an error when the button is not clicked', () => {
    // Arrange
    const { container } = render(<CreateError />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
