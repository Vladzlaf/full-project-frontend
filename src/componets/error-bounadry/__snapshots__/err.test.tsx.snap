// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateError should not throw an error when the button is not clicked 1`] = `
<div>
  <div
    class="wrap"
  >
    <button
      class="wrap__btn--err"
    >
      Click and you will see
    </button>
  </div>
</div>
`;
